src/middlewares/auth.js

```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const authentication = (req, res, next) => {
+  const bearerHeader = req.headers.authorization;
+  if (bearerHeader) {
+    const bearer = bearerHeader.split(' ');
+    const token = bearer[1];
+    jwt.verify(token, process.env.JWT_SECRET, (err, decoded) => {
+      if (err) {
+        return res.status(401).json({
+          status: 401,
+          error: 'Unauthorized! token is invalid, please sign in again'
```

Comment:
We would be verifying token for all protected endpoints and not just the sign-in endpoint thus please remove `Please sign in again` form the message. A simple message like `Token not valid` will suffice.